internal/dns/takeover.go

```diff
@@ -0,0 +1,167 @@
+package dns
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	osintscan "github.com/Method-Security/osintscan/generated/go"
+)
+
+func DetectDomainTakeover(targets []string, setHTTP bool, timeout int) (*osintscan.DomainTakeoverReport, error) {
+	resources := osintscan.DomainTakeoverReport{}
+	errs := []string{}
+
+	httpClient := createHTTPClient(setHTTP, timeout)
+
+	fingerprints, err := retrieveFingerprints()
+	if err != nil {
+		return &resources, err
+	}
+
+	var takeoverResults []*osintscan.DomainTakeover
+	for _, target := range targets {
+
+		// Fetch DNS records
+		domain, cname, err := retrieveCNAMERecord(target)
+		if err != nil {
+			errs = append(errs, err.Error())
+			continue
+		}
+
+		responseBody, statusCode, serviceResults, err := assessTarget(target, httpClient, fingerprints)
+		if err != nil {
+			errs = append(errs, err.Error())
+			continue
+		}
+		takeoverResult := osintscan.DomainTakeover{
+			Target:       target,
+			ResponseBody: responseBody,
+			StatusCode:   statusCode,
+			Domain:       domain,
+			Cname:        cname,
+			Services:     serviceResults,
+		}
+		takeoverResults = append(takeoverResults, &takeoverResult)
+	}
+
+	resources.DomainTakeovers = takeoverResults
+	resources.Errors = errs
+	return &resources, nil
+}
+
+func createHTTPClient(setHTTP bool, timeout int) *http.Client {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: !setHTTP},
+	}
+	return &http.Client{
+		Timeout:   time.Duration(timeout) * time.Second,
+		Transport: tr,
+	}
+}
+
+func retrieveFingerprints() ([]osintscan.Fingerprint, error) {
+	var fingerprints []osintscan.Fingerprint
+
+	absPath, err := filepath.Abs("configs/fingerprints.json")
```

Comment:
Should this be configurable?

Reply:
I dont think so. This file is grabbed from [repo](https://github.com/EdOverflow/can-i-take-over-xyz/blob/master/README.md) and has a very specific structure. Im in the camp that it doesn't make enable users to send in there own fingerprints for this situation

Reply:
My counterargument is that the CLI does not work then if it's not in the container

Reply:
I can add a command in the Dockerfile to pull the file from the repo. Or do you suggest just making it configurable?

Reply:
I think a configurable flag with a sane default (eg. `configs/fingerprints.json`) would be ideal. That allows the user to not _have_ to configure it if they're in the container but allows for the option.